MiniAn/minian_run.py

```diff
@@ -65,501 +29,19 @@
-
-
-params_update_temporal = {
-    'noise_freq': noise_freq,
-    'sparse_penal': parameters["TemporalPenalty"],
-    'p': 1,
-    'add_lag': 20,
-    'jac_thres': 0.2
-}
-if "update_temporal" in advanced_settings:
-    params_update_temporal, advanced_settings["update_temporal"] = mn_utils.set_advanced_parameters_for_func_params(params_update_temporal, advanced_settings["update_temporal"], update_temporal)
-
-# Update AdvancedSettings in params_doric
-parameters["AdvancedSettings"] = advanced_settings.copy()
-
 if __name__ == "__main__":
+    minian_params = mn_param.MinianParameters(danse_parameters)
 
-    # Start cluster
-    print(START_CLUSTER, flush=True)
-    cluster = LocalCluster(**params_LocalCluster)
-    annt_plugin = TaskAnnotation()
-    cluster.scheduler.add_plugin(annt_plugin)
-    client = Client(cluster)
-
-    # MiniAn CNMF
-    intpath = os.path.join(dpath, "intermediate")
-    subset = {"frame": slice(0, None)}
-
-    ### Load and chunk the data ###
-    print(LOAD_DATA, flush=True)
-    varr, file_ = mn_utils.load_doric_to_xarray(**params_load_doric)
-    chk, _ = get_optimal_chk(varr, **params_get_optimal_chk)
-    varr = save_minian(varr.chunk({"frame": chk["frame"], "height": -1, "width": -1}).rename("varr"),
-                       intpath, overwrite=True)
-    varr_ref = varr.sel(subset)
-
-    ### Pre-process data ###
-    print(PREPROCESS, flush=True)
-    # 1. Glow removal
-    print(PREPROC_REMOVE_GLOW, flush=True)
-    varr_min = varr_ref.min("frame").compute()
-    varr_ref = varr_ref - varr_min
-    # 2. Denoise
-    print(PREPROC_DENOISING, flush=True)
-    try:
-        varr_ref = denoise(varr_ref, **params_denoise)
-    except TypeError:
-        utils.print_to_intercept(ONE_PARM_WRONG_TYPE.format("denoise"))
-        sys.exit()
-    # 3. Background removal
-    print(PREPROC_REMOV_BACKG, flush=True)
-    try:
-        varr_ref = remove_background(varr_ref, **params_remove_background)
-    except TypeError:
-        utils.print_to_intercept(ONE_PARM_WRONG_TYPE.format("remove_background"))
-        sys.exit()
-    # Save
-    print(PREPROC_SAVE, flush=True)
-    varr_ref = save_minian(varr_ref.rename("varr_ref"), intpath, overwrite=True)
-
-    ### Motion correction ###
-    if parameters["CorrectMotion"]:
-        print(CORRECT_MOTION_ESTIM_SHIFT, flush=True)
-        try:
-            motion = estimate_motion(varr_ref, **params_estimate_motion)
-        except TypeError:
-            utils.print_to_intercept(ONE_PARM_WRONG_TYPE.format("estimate_motion"))
-            sys.exit()
-        motion = save_minian(motion.rename("motion").chunk({"frame": chk["frame"]}), **params_save_minian)
-        print(CORRECT_MOTION_APPLY_SHIFT, flush=True)
-        Y = apply_transform(varr_ref, motion, **params_apply_transform)
-
+    if minian_params.preview:
+        mn_preview.minian_preview(minian_params)
```

Comment:
If you have smaller functions, then the `preview()` and `main()` can be in the same file